packages/liveblocks-core/src/room.ts

```diff
@@ -2160,6 +2193,36 @@ function makeStateMachine<
     onClose(event);
   }
 
+  function getStorageStatus(): StorageStatus {
+    if (state.root !== undefined) {
+      return state.offlineOperations.size === 0
+        ? "synchronized"
+        : "synchronizing";
+    }
+
+    if (_getInitialStatePromise !== null) {
+      return "loading";
+    }
+
+    return "not-loaded";
+  }
```

Comment:
Small readability suggestion: eat all the "edge cases" first at the top, and move the happy path down:

```suggestion
  function getStorageStatus(): StorageStatus {
    if (_getInitialStatePromise === null) {
      return "not-loaded";
    }
    
    if (state.root === undefined) {
      return "loading";
    }
    
    return state.offlineOperations.size === 0
      ? "synchronized"
      : "synchronizing";
  }
```

Another nice side benefit is that it subconciously reflects the order of the state transitions that way, too.

Reply:
Yep! Way cleaner!